internal/util/netlistener/listener_holder.go

```diff
@@ -72,6 +72,9 @@ func (lh *ListenerHolder) Close() error {
 func NewFromPortNumber(portNumber int) (*ListenerHolder, error) {
 	addr := fmt.Sprintf(":%d", portNumber)
 	conn, err := net.Listen("tcp", addr)
+	if err != nil {
```

Comment:
This check collides with the if err == nil check on line 84.

Reply:
I ve removed that check - This one is required since there is a null dereference panic otherwise on 79.